### ISSUE TITLE
StorageSample: Wrong assignment of GTMSessionFetcher to GTMHTTPFetcher 

### ISSUE THREAD

**Issue by simonmeier** [google/google-api-objectivec-client#105]
In the storageSample, in downloadFileClicked the fetcherService returns a GTMSessionFetcher, where there is a GTMHTTPFetcher expected:

In file 'StorageSampleWindowsController.m', line 185:
GTMHTTPFetcher *fetcher = [self.storageService.fetcherService fetcherWithURLString:urlString];
It sais:
"StorageSampleWindowController.m:185:27: Incompatible pointer types initializing 'GTMHTTPFetcher *' with an expression of type 'GTMSessionFetcher *'"

I don't yet understand the neither content of the two fetchers, their differences and the already described issues in this issue list regarding those twos. Nevertheless I assume there is a coding error in the sample.

Regarding the described HTTP- / Session- Fetcher reference problems:
I fixed the missing files by referencing to the cloned files of the google projects gtm-session-fetcher and gtm-http-fetcher (where for the latter I added the -fno-objc-arc compiler flag). I assume that is the right fix and the problem lies, as said in the sample.

**Comment by JonasGessner** [comment]
Yeah, the code deciding whether to use the session fetcher vs the HTTP fetcher is wrong. More details on it in are here: #100
